contracts/node.sh

```diff
@@ -7,8 +7,13 @@ RED='\033[0;31m'
 NO_COLOR='\033[0m'
 
 main()
-{
-    rm -rf deployments/localhost
+{   
+    if [ "$IS_TEST" == "true" ] && [ "$FORK" == "true" ]; 
```

Comment:
Actually, I think the change to this script is not necessary. 

This line here: <https://github.com/OriginProtocol/origin-dollar/blob/shah/fork-tests-r2/contracts/fork-test.sh#L59>

The `fork-test.sh` script copies deployments from `localhost` network when using a local forked node and from `mainnet` network otherwise

Reply:
Yeah that works correctly when spinning up node via `fork-test.sh` but didn't work for me when using 2 terminal setup. Since the node is already running in the first terminal and has already read from deployment files before the `fork-test.sh` script executes.

Reply:
Thinking more on this I am not sure copying deployments from localhost to hardhat is a good idea. Since both actually fork from mainnet right? I mean hardhat network doesn't fork its state from the localhost network (both within hardhat configuration). Rather both fork mainnet. (do correct me if I am wrong)

I was under the impression: 
- hardhat network solely used for running tests in forked mode. Forks its state from mainnet
- localhost network used when running node in local - forked development mode. Forks its state from mainnet

There is smth going on where if I run `FORK=true IS_TEST=true yarn run node` it reads some deployment info from `deployments/localhost`. You can verify this by: 
- delete `deployments/hardhat` && `deployments/localhost`
- run `FORK=true IS_TEST=true yarn run node`

It will fail when trying to fetch OUSDProxy address. If you then `cp -r deployments/mainnet deployments/localhost` the running of the node will succeed. 

This leaves me under the impression that part of the deployments are being read from hardhat folder and part from the localhost folder.

Reply:
@sparrowDom I believe this is cleared now. I'm updating the readme so that it's clear, marking this as resolved. Feel free to ping me if you have any further queries regarding this